File: Python/NotesManager.py
from os import system


class NotesManager:

  def __init__(self):
    self.notes = []

  def add_note(self, note):
    self.notes.append(note)

  def print_notes(self):
    if len(self.notes) > 0:
      for i in range(len(self.notes)):
        print(str(i+1)+". "+str(self.notes[i]))
    else:
      print("There are no notes to print!")

  def remove_and_edit_note(self,index,NewNote,action):
    #action - '0' for remove note '1' for edit note
    if index.isdigit() and 0 < int(index) <= len(self.notes):
      if int(action) == 1:
        self.notes[int(index)-1] = NewNote
      else:
        del self.notes[int(index)-1]
    else:
      print("That is not a valid index.")

  def clear_notes(self):
    self.notes.clear()


def clear():
  _ = system('clear')


#Initialises Notes
Notes = NotesManager()

#Running Loop
while True:
  action = input("""Hello and Welcome to your Notes Manager!
Here are the following actions you can peform:
'add' - add a note
'remove' - remove a note
'print' - print your notes
'edit' - edit a note
'clear' - clears all notes
'exit' - exit the program
""")
  clear()
  if action.lower() == "add":
    note = input("What note would you like to add?")
    Notes.add_note(note)
  elif action.lower() == "remove":
    Notes.print_notes()
    index = input("What number note would you like to remove?")
    Notes.remove_and_edit_note(index,"",0)
  elif action.lower() == "print":
    Notes.print_notes()
  elif action.lower() == "edit":
    Notes.print_notes()
    index = input("What number note would you like to edit?")
    newNote = input("What would you like the new note to be?")
    Notes.remove_and_edit_note(index,newNote,1)
  elif action.lower() == "clear":
    Notes.clear_notes()
  elif action.lower() == "exit":
    break
  else:
    print("That is not an action.")

File: C#/NotesManager.cs
using System;
using System.Collections.Generic;

public class NotesManager
{
    private List<string> notes;

    public NotesManager()
    {
        notes = new List<string>();
    }

    public void AddNote(string note)
    {
        notes.Add(note);
    }

    public void PrintNotes()
    {
        if (notes.Count > 0)
        {
            for (int i = 0; i < notes.Count; i++)
            {
                Console.WriteLine($"{i + 1}. {notes[i]}");
            }
        }
        else
        {
            Console.WriteLine("There are no notes to print!");
        }
    }

    public void RemoveAndEditNote(string index, string newNote, int action)
    {
        int noteIndex;
        bool isValidIndex = int.TryParse(index, out noteIndex) && noteIndex > 0 && noteIndex <= notes.Count;

        if (isValidIndex)
        {
            if (action == 1)
            {
                notes[noteIndex - 1] = newNote;
            }
            else
            {
                notes.RemoveAt(noteIndex - 1);
            }
        }
        else
        {
            Console.WriteLine("That is not a valid index, please enter the number of a note contained within the list.");
        }
    }

    public void ClearNotes()
    {
        notes.Clear();
    }
}

public class Program
{
    public static void Main(string[] args)
    {
        NotesManager notes = new NotesManager();

        while (true)
        {
            Console.Clear();
            Console.WriteLine("Hello and Welcome to your Notes Manager!\nHere are the following actions you can peform:\n'add' - add a note\n'remove' - remove a note\n'print' - print your notes\n'edit' - edit a note\n'clear' - clears all notes\n'exit' - exit the program");
            string action = Console.ReadLine();
            Console.Clear();
            switch (action.ToLower())
            {
                case "add":
                    Console.WriteLine("What note would you like to add?");
                    string note = Console.ReadLine();
                    notes.AddNote(note);
                    break;
                case "remove":
                    notes.PrintNotes();
                    Console.WriteLine("What number note would you like to remove?");
                    string indexToRemove = Console.ReadLine();
                    notes.RemoveAndEditNote(indexToRemove, "", 0);
                    break;
                case "print":
                    notes.PrintNotes();
                    break;
                case "edit":
                    notes.PrintNotes();
                    Console.WriteLine("What number note would you like to edit?");
                    string indexToEdit = Console.ReadLine();
                    Console.WriteLine("What would you like the new note to be?");
                    string newNote = Console.ReadLine();
                    notes.RemoveAndEditNote(indexToEdit, newNote, 1);
                    break;
                case "clear":
                    notes.ClearNotes();
                    break;
                case "exit":
                    return;
                default:
                    Console.WriteLine("That is not an action.");
                    break;
            }
        }
    }
}
